### PR TITLE
Jump To Screen UI implementations (closed PR #754 issue resolved)

### PR DESCRIPTION
**Pull request by imfaith-hq**
//   Made major changes to the static UI screen to resemble a more dynamic UI screen.
//   The near to pixel perfection of the UI design.
//   The implementation of the UI search logic in the view model.
//   The usage of dummy user list in the 'dm_jump_to_viewmodel. dart ' file to act as the endpoint so as to test out the logic
      and the implementation of the search feature on the UI.

### CODE REVIEW

File: lib/general_widgets/custom_channel.dart [hunk 1]
```diff
@@ -7,21 +7,24 @@ class CustomChannel extends StatelessWidget {
...
-          ),
-        ],
+      height: 24,
+      width: 211,
+      child: Padding(
+        padding: const EdgeInsets.fromLTRB(5,0,0,0),
+        child: Row(
+          children: [
+            const Text('#',
+                style: TextStyle(
```
Comment (isaacadariku) [inline lib/general_widgets/custom_channel.dart:17]: Use the const shared_TextStyle from the app

File: lib/general_widgets/custom_channel.dart [hunk 2]
```diff
@@ -7,21 +7,24 @@ class CustomChannel extends StatelessWidget {
...
+                  fontWeight: FontWeight.bold,
+                  fontSize: 20,
+                )),
+            const SizedBox(width: 21.33),
+            Text(
+              text,
+              style: const TextStyle(fontSize: 14, fontWeight: FontWeight.bold),
```
Comment (isaacadariku) [inline lib/general_widgets/custom_channel.dart:24]: Here too

File: lib/general_widgets/custom_channel_stage.dart [hunk 3]
```diff
@@ -9,24 +9,27 @@ class CustomChannelStage extends StatelessWidget {
...
-          const SizedBox(width: 20),
-          Text(
-            text,
-            style: const TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
-          ),
-        ],
+            const SizedBox(width: 18),
+            Text(
+              text,
+              style: const TextStyle(fontSize: 14, fontWeight: FontWeight.bold),
```
Comment (isaacadariku) [inline lib/general_widgets/custom_channel_stage.dart:29]: Here too

File: lib/general_widgets/custom_user.dart [hunk 4]
```diff
@@ -9,25 +9,35 @@ class CustomUser extends StatelessWidget {
...
+              height: 48,
+              decoration: BoxDecoration(
+                  borderRadius: BorderRadius.circular(8),
+                  image: DecorationImage(
+                      fit: BoxFit.cover,
+                      image: AssetImage(
+                        image,
+                      ))),
+            ),
+            SizedBox(
```
Comment (isaacadariku) [inline lib/general_widgets/custom_user.dart:27]: Use the app shared ui utilities

File: lib/general_widgets/custom_user.dart [hunk 5]
```diff
@@ -9,25 +9,35 @@ class CustomUser extends StatelessWidget {
...
+              height: 8,
+            ),
+            Container(
+                height: 32,
```
Comment (isaacadariku) [inline lib/general_widgets/custom_user.dart:31]: Will this height scale on all device?

File: lib/general_widgets/custom_user_channel.dart [hunk 6]
```diff
@@ -2,37 +2,43 @@ import 'package:flutter/material.dart';
...
-        width: 379,
+        height: 24,
+        width: 211,
         child: Row(
           children: [
             Container(
               child: Image(
-                height: 34,
-                width: 34,
+                height: 24,
```
Comment (isaacadariku) [inline lib/general_widgets/custom_user_channel.dart:22]: Same here

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Make the changes using the app shared utilities, you have to many static values

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart [hunk 7]
```diff
@@ -40,4 +42,174 @@ class DmJumpToViewModel extends BaseViewModel {
...
+          list.add(index);
+        }
+      }
+
+    }
+    notifyListeners();
+  };
+
+  //getters for the view
+  TextEditingController get controller => _controller;
```
Comment (isaacadariku) [inline lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart:213]: Use stacked_form and remove the controller from the viewmodel

**Review by isaacadariku — DISMISSED** [review]

File: lib/general_widgets/custom_channel.dart (hunk 1 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_channel.dart:17]: done

File: lib/general_widgets/custom_user.dart (hunk 4 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_user.dart:27]: I don't understand what you mean boss, I checked the ui utilities and I can't understand of what i'm to use in there.

File: lib/general_widgets/custom_user.dart (hunk 5 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_user.dart:31]: done

File: lib/general_widgets/custom_user_channel.dart (hunk 6 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_user_channel.dart]: same screen-utils package used to access some properties that can't be accessed using the const textStyle

Comment (imfaith-hq) [reply lib/general_widgets/custom_user_channel.dart]: same screen-utils package used to access some properties that can't be accessed using the const textStyle

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart (hunk 7 above)
Comment (imfaith-hq) [reply lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart]: done

File: lib/general_widgets/custom_channel.dart [hunk 8]
```diff
@@ -1,27 +1,42 @@
...
+        child: Container(
+          height: 24.h,
+          width: 211.w,
+          child: Padding(
+            padding: EdgeInsets.fromLTRB(5.w,0,0,0),
+            child: Row(
+              children: [
+                Text('#',
+                    // style: AppTextStyles.heading7,
+                    style: TextStyle(
```
Comment (Quwaysim) [inline lib/general_widgets/custom_channel.dart:25]: Apply a style from AppTextStyles class

File: lib/general_widgets/custom_channel.dart [hunk 9]
```diff
@@ -1,27 +1,42 @@
...
+                      fontWeight: FontWeight.bold,
+                      fontSize: 20.sp,
+                    )
+                ),
+                SizedBox(width: 21.33.w),
+                Text(
+                  text,
+                  // style: ZuriTextStyle.mediumBold(),
+                  style: TextStyle(fontSize: 14.sp, fontWeight: FontWeight.bold),
```
Comment (Quwaysim) [inline lib/general_widgets/custom_channel.dart:34]: Apply a style from AppTextStyles class

File: lib/general_widgets/custom_channel_stage.dart [hunk 10]
```diff
@@ -8,25 +9,34 @@ class CustomChannelStage extends StatelessWidget {
...
+                  width: 16.w,
+                  child: Icon(
+                    leadingIcon,
+                    size: 16.sp,
+                  ),
+                ),
+                SizedBox(width: 18.w),
+                Text(
+                  text,
+                  style: TextStyle(fontSize: 14.sp, fontWeight: FontWeight.bold),
```
Comment (Quwaysim) [inline lib/general_widgets/custom_channel_stage.dart:34]: Apply a style from AppTextStyles class

File: lib/general_widgets/custom_user_channel.dart [hunk 11]
```diff
@@ -1,45 +1,58 @@
...
+                    height: 24.h,
+                    width: 24.w,
+                    fit: BoxFit.cover,
+                    image: AssetImage(image),
+                  ),
+                ),
+                SizedBox(width: 12.w),
+                Text(
+                  text,
+                  style: TextStyle(fontSize: 15.sp, fontWeight: FontWeight.bold),
```
Comment (Quwaysim) [inline lib/general_widgets/custom_user_channel.dart:36]: Apply a style from AppTextStyles class

File: lib/general_widgets/custom_user_channel.dart [hunk 12]
```diff
@@ -1,45 +1,58 @@
...
+                ),
+                SizedBox(width: 10.w),
+                Text(
+                  text2,
+                  style: TextStyle(fontSize: 15.sp, fontWeight: FontWeight.normal),
```
Comment (Quwaysim) [inline lib/general_widgets/custom_user_channel.dart:41]: Apply a style from AppTextStyles class

**Review by Quwaysim — CHANGES_REQUESTED** [review]

File: lib/general_widgets/custom_channel.dart (hunk 8 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_channel.dart]: done

File: lib/general_widgets/custom_channel_stage.dart (hunk 10 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_channel_stage.dart]: done

File: lib/general_widgets/custom_user_channel.dart (hunk 11 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_user_channel.dart]: done

File: lib/general_widgets/custom_user_channel.dart (hunk 12 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_user_channel.dart]: done

File: lib/general_widgets/custom_user_channel.dart [hunk 13]
```diff
@@ -1,45 +1,59 @@
...
+                  child: Image(
+                    height: 24.h,
+                    width: 24.w,
+                    fit: BoxFit.cover,
+                    image: AssetImage(image),
+                  ),
+                ),
+                SizedBox(width: 12.w),
+                Text(
+                  text,
```
Comment (Quwaysim) [inline lib/general_widgets/custom_user_channel.dart:36]: textData here can be null and twill crash the app. Please set a default  value, return an empty string if null or make its field  "required"

File: lib/general_widgets/custom_user_channel.dart [hunk 14]
```diff
@@ -1,45 +1,59 @@
...
+                  style: AppTextStyles.fileName, // textfont in here won't scale to different screens
+                ),
+                SizedBox(width: 10.w),
+                Text(
+                  text2,
```
Comment (Quwaysim) [inline lib/general_widgets/custom_user_channel.dart:41]: Same here

File: lib/ui/nav_pages/home_page/home_page.dart [hunk 15]
```diff
@@ -36,10 +36,10 @@ class HomePage extends StatelessWidget {
           ),
           model.isBusy
               ? LinearProgressIndicator(
-                  backgroundColor: Colors.grey[400],
-                  valueColor:
-                      const AlwaysStoppedAnimation(AppColors.zuriPrimaryColor),
-                )
+            backgroundColor: Colors.grey[400],
```
Comment (Quwaysim) [inline lib/ui/nav_pages/home_page/home_page.dart:39]: Use the AppColors class

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart [hunk 16]
```diff
@@ -18,6 +19,7 @@ class DmJumpToViewModel extends FormViewModel {
   void setFormStatus() {}
 
   final log = getLogger('DmJumpToViewModel');
+
   TextEditingController _controller = TextEditingController();
```
Comment (Quwaysim) [inline lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart:23]: Remove both TextEditingControllers... If you don't know how to, check filled stacks.com

**Review by Quwaysim — CHANGES_REQUESTED** [review]

**Review by Quwaysim — APPROVED** [review]

File: lib/general_widgets/custom_user_channel.dart (hunk 13 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_user_channel.dart]: done, and its just a template for a user coming from the endpoints,

File: lib/general_widgets/custom_user_channel.dart (hunk 14 above)
Comment (imfaith-hq) [reply lib/general_widgets/custom_user_channel.dart]: done

File: lib/ui/nav_pages/home_page/home_page.dart (hunk 15 above)
Comment (imfaith-hq) [reply lib/ui/nav_pages/home_page/home_page.dart:39]: i'm not working on home_page, i only resolved conflict in it, so I could push my commits

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart (hunk 16 above)
Comment (imfaith-hq) [reply lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart]: done

**Review by Quwaysim — APPROVED** [review]

**Review by mariamadebolahamzat — APPROVED** [review]
All changes made

**Pull request merged**